warrior/Framework/ClassUtils/testdata_class.py

```diff
@@ -573,23 +587,31 @@ def resolve_iteration_patterns(self, details_dict):
                 # move on to the next command
                 # This also means that the command parameters do not have any
                 # iteration patterns because the validation was already done.
-                if cmd_iter_pattern != "":
-                    # if cmd_iterpattern is not ""
-                    # call the expand cmd_params method to resolve the
-                    # iteration patterns in the command and get a updated
+                if isinstance(cmd, str) and '+' not in cmd:
+                    # check if list exists in cmd
+                    td_obj = TestData()
```

Comment:
why intialize a new TestData object here, please use self.list_substitution_precheck

Reply:
@satyakrish 
list_substitution_precheck is in TestData Class and I am using it in TestDataIterations Class, So creating a new object for it.